hashring/rbtree.go

```diff
@@ -259,11 +247,42 @@ func (n *RBNode) search(val int) (string, bool) {
 	return "", false
 }
 
-// Search searches for a value in the RBTree, returns the string and true if found,
-// or the empty string and false if val is not in the tree
-func (t *RBTree) Search(val int) (string, bool) {
+// Search searches for a value in the RedBlackTree, returns the string and true
+// if found or the empty string and false if val is not in the tree.
+func (t *RedBlackTree) Search(val int) (string, bool) {
 	if t.root == nil {
 		return "", false
 	}
 	return t.root.search(val)
 }
+
+// LookupNAt iterates through the tree from the node with value val, and
+// returns the next n unique strings. This function is not guaranteed to
+// return n strings.
+func (t *RedBlackTree) LookupNAt(n int, val int, unique map[string]bool) {
+	dfs(t.root, n, val, unique)
+}
+
+// dfs is a depth-first-search that finds n unique strings
+// with a value bigger or equal than val
+func dfs(node *Node, n int, val int, unique map[string]bool) {
```

Comment:
What do you mean with returning multiple values, the function doesn't return any value?

We can name it searchNUniqueBiggerThan. I think I'll swap the val and the n parameters as well.